### ISSUE TITLE
Named query params with an empty array cannot be parsed

### ISSUE THREAD

**Issue by dinvlad** [googleapis/nodejs-bigquery#400]
#### Environment details

  - OS: High Sierra (10.13.6)
  - Node.js version: 8.15.1
  - npm version: 6.9.0
  - `@google-cloud/bigquery` version: 2.1.0

#### Steps to reproduce

  1. Run a query with an empty array as the value of a named query param, e.g.
```ts
const [rows] = await bq.query({
  query,
  params: {
    key: [],
  },
});
```
  2. Query fails with `This value could not be translated to a BigQuery data type`.
  3. Attempt to run the same query with an empty array literal in BigQuery UI or CLI and observe that the query succeeds:
```sql
...
WHERE `dataset.table`.Key IN UNNEST([])
```

#### Source of the error
I believe the error is a result of the login in https://github.com/googleapis/nodejs-bigquery/blob/ba5c82efc4c6f2aa50df2c386ab0b6894d0dd2aa/src/index.ts#L726

You can see that `nodejs-bigquery` is trying to determine the type of the query param from the first element. If the array is empty, `value[0]` is `undefined`, and `getType_()` throws an error. This issue is similar to https://github.com/googleapis/google-cloud-java/issues/2678, where they did fix it (and as they state there, it is not a limitation of the API).

EDIT: The API does seem to require specifying `arrayType`, e.g.
```json
"parameterType": {
  "type": "ARRAY",
  "arrayType": {
    "type": "INT64"
  }
},
```
Since JavaScript (unlike Java) doesn't store the array type info at runtime (afaik), I wonder if there's a way to supply the type explicitly in this case. Not sure if this is more of a feature request and would you consider a PR for it.

Thanks!

**Comment by callmehiphop** [comment]
@stephenplusplus I'm actually pretty surprised this hasn't come up before, have you ever ran across any similar issues? I was looking for some kind of `types` option, but I didn't see any mention of one. This definitely seems like something we want.

**Comment by sduskis** [comment]
@bcoe, do you plan on fixing this?  If not, please unassign yourself.

**Comment by JustinBeckwith** [comment]
@callmehiphop giving this a bump.  Could I ask you to take a look here?

**Comment by callmehiphop** [comment]
@JustinBeckwith I'm changing this to a FR but I'm going to take it on next

**Comment by meredithslota** [comment]
@callmehiphop It seems like this is fixed w/ #4271; can you or @steffnay confirm?
